### ISSUE TITLE
Integrate with external git repository

### ISSUE THREAD

**Issue by bienchu** [curiefense/curiefense#565]
**Is your feature request related to a problem? Please describe.**
Currently, curiefense manage an internal git repository as policy control configuration store. However, in most cases, users might have their own git management tools like Gitlab / Bitbucket. In order to allow users manage ALL git repositories in a central place, curiefense should support integration with external git repository. Also, managing policy out of curiefense can make it closer to stateless.

**Comment by jdorfman** [comment]
@bienchu you are absolutely right. This will bring true GitOps into Curiefense. I'll put it on the team's radar. =)

**Comment by sagikazarmark** [comment]
Have you made any progress on this?

I can see that the confserver image can actually pull from a remote repository [here](https://github.com/curiefense/curiefense/blob/main/curiefense/images/confserver/bootstrap/bootstrap_config.sh#L17).

But it doesn't seem to be trivial to set that repo up manually and I'm also not sure if the confserver would push changes back to the remote.

Being able to use a central Git repository as a single source of truth seems vital from a GitOps perspective.

**Comment by tzuryby** [comment]
@xavier-rbz can we prioritize this one?
Don't seem like big of a hassle to me.

**Comment by xavier-rbz** [comment]
Current status:
* APIs for fetching and pulling exist in confserver
* added support for git operations using an SSH private key: #1162 
* added support for SSH private key in charts: https://github.com/curiefense/curiefense-helm/pull/34 

Possible next steps, depending on product design choices:
* Add support in the UI?
* Make confserver push to a remote after each operation?
* Add a script that calls the fetch & push APIs periodically?

**Comment by sagikazarmark** [comment]
From my perspective, the requirement is to make sure that the configuration does not get lost if something happens with the cluster.

I could probably use Kubernetes persistence if I wanted to, but I'd rather avoid that, since the config is already in a Git repo.
